src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -1121,5 +1125,318 @@ private String getExistingBoxFolderIdFromBoxException(BoxAPIException e,
 		}
 		return existingFolderId;
 	}
+	
+	
+	/**************** MailArchive content ***********/
+	/**
+	 * Download MailArchive resource in zip file
+	 * @param env
+	 * @param request
+	 * @param response
+	 * @param userId
+	 * @param sessionAttributes
+	 * @param site_id
+	 * @param migrationId
+	 * @param repository
+	 */
+	public void downloadMailArchiveZipFile(Environment env, HttpServletRequest request,
+			HttpServletResponse response, String userId,
+			HashMap<String, Object> sessionAttributes, String site_id,
+			String migrationId, MigrationRepository repository) {
+		// hold download status
+
+		JSONObject downloadStatus = new JSONObject();
+
+		// login to CTools and get sessionId
+		if (sessionAttributes.containsKey(Utils.SESSION_ID)) {
+			String sessionId = (String) sessionAttributes.get(Utils.SESSION_ID);
+			HttpContext httpContext = (HttpContext) sessionAttributes
+					.get("httpContext");
+			try {
+				//
+				// Sends the response back to the user / browser. The
+				// content for zip file type is "application/zip". We
+				// also set the content disposition as attachment for
+				// the browser to show a dialog that will let user
+				// choose what action will he do to the sent content.
+				//
+				response.setContentType("application/zip");
```

Comment:
`application/zip` this should be made constant as this a used when zipping resources as well